docs/content/guides/dagster/local_to_production_tutorial.mdx

```diff
@@ -0,0 +1,395 @@
+
+```python file=/guides/dagster/local_to_prod_tutorial/resources/resources_v2.py startafter=start_abc endbefore=end_abc
+# resources.py
+
+class HNClient(ABC):
+    """
+    Base class for a Hacker News Client
+    """
+
+    @property
+    def item_field_names(self):
+        pass
+
+    def fetch_item_by_id(self, item_id: int) -> Optional[Dict[str, Any]]:
+        pass
+
+    def fetch_max_item_id(self) -> int:
+        pass
+```
+
+then we can write the version of the resource we will use for testing:
+
+```python file=/guides/dagster/local_to_prod_tutorial/resources/resources_v2.py startafter=start_mock endbefore=end_mock
+# resources.py
+
+class MockHNClient(HNClient):
+    """
+    Hacker News Client that returns mock data
+    """
+
+    def __init__(self):
+        self.data = {
+            1: {
+                "id": 1,
+                "type": "comment",
+                "title": "the first comment",
+                "by": "user1",
+            },
+            2: {
+                "id": 2,
+                "type": "story",
+                "title": "an awesome story",
+                "by": "user2",
+            },
+        }
+
+    @property
+    def item_field_names(self):
+        return ["id", "type", "title", "by"]
+
+    def fetch_item_by_id(self, item_id: int) -> Optional[Dict[str, Any]]:
+        return self.data.get(item_id)
+
+    def fetch_max_item_id(self) -> int:
+        return len(self.data.items())
+
+
+@resource
+def mock_hn_client():
+    return MockHNClient()
+```
+
+Now, let’s write tests to verify that the assets produce the datasets we expect:
+
+```python file=/guides/dagster/local_to_prod_tutorial/test_assets.py startafter=start endbefore=end
+# test_assets.py
+
+def test_items():
+    context = build_op_context(resources={"hn_client": mock_hn_client})
+    hn_dataset = items(context)
+    assert isinstance(hn_dataset, pd.DataFrame)
+
+    expected_data = pd.DataFrame(MockHNClient().data.values()).rename(columns={"by": "user_id"})
+
+    assert (hn_dataset == expected_data).all().all()
+
+
+def test_comments():
+    mock_data = pd.DataFrame({"id": [1, 2, 3], "type": ["comment", "story", "comment"]})
+    out_data = comments(mock_data)
+
+    out_ids = out_data["id"].tolist()
+
+    assert 1 in out_ids
+    assert 2 not in out_ids
+    assert 3 in out_ids
+
+
+def test_stories():
+    mock_data = pd.DataFrame({"id": [1, 2, 3], "type": ["story", "story", "comment"]})
+    out_data = stories(mock_data)
+
+    out_ids = out_data["id"].tolist()
+
+    assert 1 in out_ids
+    assert 1 in out_ids
+    assert 3 not in out_ids
+```
+
+### Part Three: Production
+
+Now that our assets work locally and can be tested, let’s set up for the production environment. We want to store the assets in a production Snowflake database, so we need to update the configuration for the `snowflake_io_manager`:
```

Comment:
> Now that our assets work locally and can be tested, let’s set up for the production environment.

I'm finding the second phrase in this sentence a bit awkward. Maybe something like `... and can be tested, let's modify our code for a production environment.`

Open to suggestions!